core/src/main/java/dk/alexandra/fresco/framework/builder/numeric/AdvancedNumeric.java

```diff
@@ -127,6 +129,20 @@
   DRes<SInt> bitLength(DRes<SInt> input, int maxBitLength);
 
   DRes<SInt> invert(DRes<SInt> x);
+  
+  /**
+   * Selects left or right based on condition.
+   *
+   * @param condition the Computation holding the condition on which to select.
+   * @param left the Computation holding the left argument.
+   * @param right the Computation holding the right argument.
+   * @return an computation holding the output of the appended protocol.
+   */
+  DRes<SInt> condSelect(DRes<SInt> condition, DRes<SInt> left,
+      DRes<SInt> right);
+  
+  DRes<Pair<DRes<SInt>, DRes<SInt>>> condSwap(DRes<SInt> condition, DRes<SInt> left,
```

Comment:
It would be nice to have javadoc here

Reply:
An idea: Maybe it makes more sense to coin these two method:
- selectIf
- swapIf

?

Reply:
Agree with swapIf, don't like selectIf. To me selectIf implies that something will be selected if the condition is true and nothing else will happen otherwise.